packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts

```diff
@@ -175,4 +192,108 @@ describeNoCompat("GC loading from older summaries", (getTestObjectProvider) => {
+        const unreferencedTimestamps3 = await getUnreferencedTimestamps(summaryResult3.summaryTree);
+        const dsBTime3 = unreferencedTimestamps3.get(dataStoreB._context.id);
+        assertNotTrue(dsBTime3 === dsBTime2, `dataStoreB's time should be same as in summary2`);
+    });
+
+    it("does not log gcUnknownOutboundReferences errors when loading from an older summary", async () => {
+        const summarizer1 = await createSummarizer(provider, mainContainer);
+
+        // Create a data store and mark it unreferenced to begin with.
+        const dataStoreBHandle =
+            (await containerRuntime.createDataStore(TestDataObjectType)).entryPoint as IFluidHandle<ITestDataObject>;
+        assert(dataStoreBHandle !== undefined, "New data store does not have a handle");
+        const dataStoreB = await dataStoreBHandle.get();
+        dataStoreA._root.set("dataStoreB", dataStoreBHandle);
+        dataStoreA._root.delete("dataStoreB");
+
+        await provider.ensureSynchronized();
+
+        // Summarize - summary1. dataStoreB should be unreferenced.
+        const summaryResult1 = await summarizeNow(summarizer1);
+        const referenceState1 = await getReferenceState(summaryResult1.summaryTree);
+        const dsAReferenceState1 = referenceState1.get(dataStoreA._context.id);
+        assert(dsAReferenceState1 === true, `dataStoreA should be referenced`);
+        const dsBReferenceState1 = referenceState1.get(dataStoreB._context.id);
+        assert(dsBReferenceState1 === false, `dataStoreB should be unreferenced`);
+
+        // Reference dataStoreB. This should result in an explicit reference from dataStoreA -> dataStoreB.
+        dataStoreA._root.set("dataStoreB", dataStoreB.handle);
+
+        // Summarize - summary2. dataStoreB should now be referenced.
+        await provider.ensureSynchronized();
+        const summaryResult2 = await summarizeNow(summarizer1);
+        const unreferencedTimestamps2 = await getUnreferencedTimestamps(summaryResult2.summaryTree);
+        const dsBTime2 = unreferencedTimestamps2.get(dataStoreB._context.id);
+        assert(dsBTime2 === undefined, `dataStoreB's time should have updated`);
+
+        // Load a new summarizer from the summary1 and summarize - summary3. Before it summarizes, it will catch up
+        // to latest and so the reference state of the data stores should be the same as in summary2.
+        // Also, note that while catching up, it will download summary2 and update state from it.
+        summarizer1.close();
+        const summarizer2 = await createSummarizer(provider, mainContainer, summaryResult1.summaryVersion);
+        await provider.ensureSynchronized();
+
+        // When GC runs as part of this summarize, it should not throw "gcUnknownOutboundReferences" error for the
+        // dataStoreA -> dataStoreB route.
+        const summaryResult3 = await summarizeNow(summarizer2);
```

Comment:
Should we add a temporary `assertNotTrue` checking to see that `gcUnknownOutboundReferences` isn't even logged? I assume it's logged as a generic event

Reply:
It's not logged and that's why its not part of the "ifExpects" block. It will start showing up after this PR is merged which is hiding the bug in this scenario - https://github.com/microsoft/FluidFramework/pull/13361

Reply:
I will update this PR once #13361 is merged.

Reply:
Oh yeah, that's right, why do we still see `fluid:telemetry:Summarizer:Running:gcUnknownOutboundReferences` today then?